### PR TITLE
fix: if there is a symbol '?' in the path, force url. query as the mi…

### LINKED ISSUE #3013: Download request for Kserve custom resource model that does not recognize HTTP
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
After I created a custom resource through the InferenceService, I provided a correct model link that can be downloaded through HTTP. The Pod created by the custom resource will prompt the following error: "RuntimeError: URI: http://bw-model-svc.bluewhale-platform:8001/modelservice/model/download?path=/bluewhale/model/74/00000123.zip Did not respond with 'Content Type': 'application/octet stream'

**What did you expect to happen:**
I hope this link can be correctly recognized by Kserve

**What's the InferenceService yaml:**
[To help us debug please run `kubectl get isvc $name -n $namespace -oyaml` and paste the output]
```
apiVersion: serving.kserve.io/v1beta1
kind: InferenceService
metadata:
  annotations:
    serving.kserve.io/deploymentMode: RawDeployment
  finalizers:
    - inferenceservice.finalizers
  name: bw-model-api-4051
  namespace: bluewhale-platform
spec:
  predictor:
    model:
      modelFormat:
        name: Tensorflow
      name: ''
      resources: {}
      storageUri: >-
        http://bw-model-svc.bluewhale-platform:8001/modelservice/model/download?path=/bluewhale/model/74/00000123.zip
```
**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- KServe Version: v0.10.0

### PR DESCRIPTION
**Pull request by wjp-cn**
…metype

**What this PR does / why we need it**:
When our HTTP based model download path contains the symbol '?', the subsequent string will be parsed into a query, causing the mimetype to be unable to obtain the correct Content Type, resulting in errors in the entire model download process.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3013 

**Type of changes**
Please delete options that are not relevant.

- [√] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:
Pass the test for both the presence and absence of the symbol '?' in the HTTP path

Please describe the tests that you ran to verify your changes and relevent result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
Model download path：'http://172.16.1.129:9999/00000123.zip'
- [ ] Test B
Model download path：'http://172.16.2.132:32372/modelservice/model/download?path=bluewhale/model/74/00000123.zip'
- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [√] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

### CODE REVIEW

File: python/kserve/kserve/storage/storage.py [hunk 1]
```diff
@@ -525,7 +525,11 @@ def _download_local(uri, out_dir=None):
     def _download_from_uri(uri, out_dir=None):
         url = urlparse(uri)
         filename = os.path.basename(url.path)
-        mimetype, encoding = mimetypes.guess_type(url.path)
+        # Determine if the symbol '?' exists in the path
+        if  mimetypes.guess_type(url.path)[0] is None:
```
Comment (yuzisun) [inline python/kserve/kserve/storage/storage.py]: Also check `url.query != ''` ?

Comment (wjp-cn) [reply python/kserve/kserve/storage/storage.py]: Yes, I overlooked this check and added it now

Comment (yuzisun) [reply python/kserve/kserve/storage/storage.py]: Can you also add a test case here
https://github.com/kserve/kserve/blob/master/python/kserve/kserve/storage/test/test_storage.py

Comment (wjp-cn) [reply python/kserve/kserve/storage/storage.py]: Of course, there is indeed a need for a test case here

**Comment by yuzisun** [comment]
Thanks @Wercurial !

/lgtm
/approve

**Pull request merged**